src/python/WMCore/WMException.py

```diff
@@ -114,6 +116,13 @@ def addInfo(self, **data):
         exception instance
         """
         for key, value in viewitems(data):
+            # unicode sandwich: convert sequence of bytes strings to
+            # unicode codepoints strings as soon as possible
+            # assumption: value is not iterable (list, dict, tuple, ...)
+            if isinstance(key, bytes):
+                key = key.decode("utf-8", "ignore")
+            if isinstance(value, bytes):
+                value = value.decode("utf-8", "ignore")
             self[key] = value
```

Comment:
```suggestion
            self.data[key] = value
```
And here we have a hard bug! I think the code I suggest here is the correct one.

Reply:
You forsee the future, that is the topic of #10168

Reply:
Here is my comment on this exact line: https://github.com/dmwm/WMCore/pull/10168#issuecomment-743386445

Reply:
(We continued this conversation on slack. we are not including this change in this PR, we are postponing this to #10168 )